client_plugin/drivers/shared/shared_driver.go

```diff
@@ -111,7 +148,114 @@ func (d *VolumeDriver) UnmountVolume(name string) error {
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+
+	// Initialize volume metadata in KV store
+	hostname, err := os.Hostname()
+	if err != nil {
+		msg = fmt.Sprintf("Failed to fetch hostname during volume creation")
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+	hnames = append(hnames, hostname)
+	volRecord := VolumeMetadata{
+		status:         volStateCreating,
+		globalRefcount: 0,
+		port:           0,
+		username:       "root",
+		password:       "badpass",
+		clientList:     hnames,
+	}
+	log.Infof("Attempting to write initial metadata entry for %s", r.Name)
+	err = d.writeVolMetadata(r.Name, volRecord)
+	if err != nil {
+		msg = fmt.Sprintf("Failed to create volume %s. Reason: %v",
+			r.Name, err)
+		log.Warningf(msg)
+		return volume.Response{Err: msg}
+	}
+
+	// Create traditional volume as backend to shared volume
+	msg = fmt.Sprintf("Attempting to create internal volume for %s", r.Name)
+	log.Infof(msg)
+	internalVolname := tradVolumePrefix + r.Name
+	dockerVolOptions := dockerTypes.VolumeCreateRequest{
+		Driver:     d.sharedDriverSlave,
+		Name:       internalVolname,
+		DriverOpts: r.Options,
+	}
+	_, err = d.dockerd.VolumeCreate(context.Background(), dockerVolOptions)
+	if err != nil {
+		omsg := fmt.Sprintf("Failed to create volume %s. Reason: %v",
+			r.Name, err)
+		log.Warningf(omsg)
+
+		// If failed, attempt to delete the metadata for this volume
+		err = d.deleteVolMetadata(r.Name)
+		if err != nil {
+			msg = fmt.Sprintf("%v. Please remove metadata entry manually using etcd CLI.", err)
+			log.Warningf(msg)
+			omsg = omsg + msg
+		}
+		return volume.Response{Err: omsg}
+	}
+
+	// TODO Set a watcher for this volume
+	log.Infof("Attempting to set watcher for volume: %s", r.Name)
+
+	// Update metadata to indicate successful volume creation
+	log.Infof("Attempting to update volume state to ready for volume: %s", r.Name)
+	volRecord = VolumeMetadata{
+		status:         volStateReady,
+		globalRefcount: 0,
+		port:           0,
+		username:       "root",
+		password:       "badpass",
+		clientList:     hnames,
+	}
+	err = d.writeVolMetadata(r.Name, volRecord)
+	if err != nil {
+		omsg := fmt.Sprintf("Failed to create volume %s. Reason: %v", r.Name, err)
+		log.Warningf(omsg)
+
+		// If failed, attempt to remove the backing trad volume
+		log.Infof("Attempting to delete internal volume")
+		err = d.dockerd.VolumeRemove(context.Background(), internalVolname)
+		if err != nil {
+			msg = fmt.Sprintf(" Failed to remove internal volume. Please remove")
+			msg += fmt.Sprintf("this volume manually using docker cli. Reason: %v ", err)
+			log.Warningf(msg)
+			omsg = omsg + msg
+		}
+
+		// Attempt to delete the metadata for this volume
+		err = d.deleteVolMetadata(r.Name)
+		if err != nil {
+			// Ques: Do we print etcddtl (cli) command here as corrective action?
+			msg = fmt.Sprintf(". Please remove metadata entry manually using etcd CLI.")
```

Comment:
I certainly saw this already 

Also, please do not use "ques" , put TODO instead (easier to search for)